### ISSUE TITLE
attention_weights_* missing 1/sqrt(head_dim) scaling

### ISSUE THREAD

**Issue by TidalPaladin** [TidalPaladin/vit#82]
The attention-weight helper functions in `vit/attention.py` compute:

```
(q @ k.mT).softmax(dim=-1)
```

without the `1/sqrt(head_dim)` scaling used by `scaled_dot_product_attention`. This makes these weights inconsistent with the actual attention probabilities from the forward pass.

Proposed fix:
- Multiply logits by `1 / sqrt(head_dim)` before softmax in:
  - `attention_weights_qkv_packed`
  - `attention_weights_q_kv_packed`
  - `attention_weights_q_kv_packed_static_query`
- Add a small test comparing these weights to `scaled_dot_product_attention` on the same inputs.
